api/v1alpha4/azuremachine_validation.go

```diff
@@ -132,27 +130,23 @@ func ValidateOSDisk(osDisk OSDisk, fieldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Required(fieldPath.Child("OSType"), "the OS type cannot be empty"))
 	}
 
-	allErrs = append(allErrs, validateStorageAccountType(osDisk.ManagedDisk.StorageAccountType, fieldPath)...)
-
 	allErrs = append(allErrs, validateCachingType(osDisk.CachingType, fieldPath)...)
 
-	if errs := ValidateManagedDisk(osDisk.ManagedDisk, osDisk.ManagedDisk, fieldPath.Child("managedDisk")); len(errs) > 0 {
-		allErrs = append(allErrs, errs...)
+	if osDisk.ManagedDisk != nil {
+		if errs := validateManagedDisk(osDisk.ManagedDisk, fieldPath.Child("managedDisk")); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
 	}
 
-	if err := validateDiffDiskSetings(osDisk.DiffDiskSettings, fieldPath.Child("diffDiskSettings")); err != nil {
-		allErrs = append(allErrs, err)
+	if osDisk.ManagedDisk != nil && osDisk.ManagedDisk.StorageAccountType == string(compute.StorageAccountTypesUltraSSDLRS) {
```

Comment:
can we move this to `validateStorageAccountType` and make it accept the diskType as a parameter?